### PR TITLE
feat(merge-guard): admin-bypass fact for GitHub ruleset review-count rejections

### PR DESCRIPTION
**Pull request by scotthamilton77**
## Summary
- Adds `facts.admin_bypass` to `check-merge-eligibility.sh` — reports whether the currently-authenticated `gh` identity already holds a standing GitHub bypass grant on this branch's required-approving-review ruleset rule. Never a blocker; eligibility is untouched.
- Documents the `--admin` retry procedure in `merge-guard/SKILL.md` Step 5 and in `docs/architecture/review-merge-policy/design.md`: usable only after this policy's own eligibility + Axis 2 authorization already held, and only once GitHub's rejection text is read and confirmed to name the approving-review requirement — `--admin` is a blanket bypass of the whole covering ruleset, not scoped to the review rule alone, so the docs are explicit about that scope and about never inferring the rejection reason.
- Fixes the GitHub bypass enum used in docs/tests (`never` → the real `none`) and adds two edge-case tests (unrecognized bypass value, null `ruleset_id`).

## Context for reviewers
This closes the gap discovered while merging PR #209: `main`'s ruleset requires 1 approving review, but Copilot's review state is never `APPROVED` and the PR author can't self-approve — so `gh pr merge` gets refused even when this repo's own two-axis review/merge policy has already authorized the merge. Investigation confirmed via `gh api` that the ruleset already grants `current_user_can_bypass: always` to the repo-admin role; GitHub just requires that grant to be exercised explicitly per merge via `--admin` rather than applying it implicitly. The fix detects that grant as a fact rather than hard-coding `--admin` everywhere, and the docs are deliberately conservative about when it's safe to use.

**Scope note:** two related-but-out-of-scope items surfaced during review and are filed separately, not fixed here: the CI-eligibility floor's blindness to ruleset-defined (as opposed to classic-protection) required status checks (`agents-config-7d5h8`), and a DRY opportunity between the new fact's 404-tolerant fetch and the existing CI-green block's fetch (`agents-config-oh7a7`). Both are `discovered-from` this PR's parent bead, `agents-config-t7esw`.

## Test plan
- [x] `bash -n` clean on both changed scripts
- [x] `check-merge-eligibility_test.sh`: 110/110 assertions passing (98 pre-existing + 12 new)
- [x] `quality-reviewer` agent pass, verified live against this repo's actual ruleset API — findings addressed
- [x] `simplify` pass (3 parallel reviewers: reuse/quality/efficiency) — no auto-apply items; one flagged duplication deferred with rationale

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/user/.agents/skills/merge-guard/check-merge-eligibility.sh [hunk 1]
```diff
@@ -267,6 +272,49 @@ if [[ "$ci_state" != "green" && "$ci_state" != "none" ]]; then
...
+if rules_json=$(gh api "repos/${OWNER}/${REPO}/rules/branches/${BASE_REF_ENC}" 2>"$rules_stderr"); then
+    :
+else
+    if grep -qiE 'HTTP 404|Not Found' "$rules_stderr"; then
+        rules_json='[]'   # no rulesets target this branch
+    else
+        echo "Error: failed to fetch branch rules: $(cat "$rules_stderr")" >&2
+        rm -f "$rules_stderr"; exit 3
+    fi
+fi
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/merge-guard/check-merge-eligibility.sh:298]: Fixed in f98f9b75d2e9ee6148529a778f6523f2b619d9b7. List-level rules/branches fetch: on a non-404 failure, warns on stderr and degrades rules_json to '[]' (same inert path as a legitimate 404) instead of exit 3. Per-ruleset detail fetch: on failure, warns, sets current_actor_can_bypass=false (fail-closed default for the narrow --admin decision), and continues instead of aborting. admin_bypass never gates blockers[] (set_fact, never add_blocker), so a transient fetch failure on it must not deny a merge that may not even need --admin. The pre-existing CI-green block's exit 3 (which guards a real blocker) was left untouched.

File: src/user/.agents/skills/merge-guard/check-merge-eligibility.sh [hunk 2]
```diff
@@ -267,6 +272,49 @@ if [[ "$ci_state" != "green" && "$ci_state" != "none" ]]; then
...
+current_actor_can_bypass=null
+if [[ "$required_approving_review_count" -gt 0 ]]; then
+    review_rule_active=true
+    current_actor_can_bypass=true
+    ruleset_ids=$(jq -r '[ .[].ruleset_id ] | unique | .[]' <<<"$review_rulesets")
+    while IFS= read -r rid; do
+        [[ -n "$rid" ]] || continue
+        bypass=$(gh_api "repos/${OWNER}/${REPO}/rulesets/${rid}" --jq '.current_user_can_bypass') || {
+            echo "Error: failed to fetch ruleset ${rid} bypass status" >&2; exit 3; }
+        [[ "$bypass" == "always" || "$bypass" == "pull_requests_only" ]] || current_actor_can_bypass=false
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply src/user/.agents/skills/merge-guard/check-merge-eligibility.sh:318]: Fixed via the change addressing https://github.com/scotthamilton77/agents-config/pull/210#discussion_r3516810278.

**Comment by scotthamilton77** [comment]
This is the review's auto-generated PR overview describing the diff, not an actionable request — no changes needed.

**Comment by scotthamilton77** [comment]
This is the re-review's auto-generated overview confirming no new comments were generated — not an actionable request.
